activerecord/lib/active_record/autosave_association.rb

```diff
@@ -388,8 +388,10 @@ def save_collection_association(reflection)
               records -= records_to_destroy
             end
 
+            is_habtm = (reflection.parent_reflection && reflection.parent_reflection.macro == :has_and_belongs_to_many)
+
             records.each do |record|
-              next if record.destroyed?
+              next if record.destroyed? || (is_habtm && record.persisted? && !record.changed?)
```

Comment:
The foreign_key will already be set at this point, right?

If so, then we can remove `is_habtm` and just go with `record.destroyed? || (record.persisted? && !record.changed?)`, since the fact that the FK is set means changed will be true if it's a new record.